### PR TITLE
node syncing and weak subjectivity

### PR DESCRIPTION
**Pull request by chirag-parmar**
* Adds a new section to the consensus layer that addresses weak subjectivity

TODO:
* [x] add links to resources for weak subjectivity
* [x] add the section to the nav bar 
* [x] remove mentions of ETH1 from the doc
* [x] make a clearer distinction b\w sync strategies of the EL and the CL
* [x] highlight the role of finality in weak subjectivity

### CODE REVIEW

File: docs/wiki/CL/syncing.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+With the merge, execution blocks are wrapped inside beacon blocks (the beacon chain existed in parallel to the main chain, hence "The Merge") and clients are split into three, an execution layer client, a consensus layer client, and optionally a validator client. Due to this change, the CL node depends on the EL node to verify the execution payload within the beacon blocks. For syncing, however, the EL node depends on the CL node for a sync target. It then follows the same steps as an ETH1 client except. To provide this target, the CL node needs to sync the beacon chain first and it ***CANNOT*** use the same approach as the EL node, i.e., to simply sync from genesis. In fact, if you have ever run a CL node, it would complain that syncing from genesis is [unsafe](https://docs.teku.consensys.io/concepts/weak-subjectivity#sync-outside-the-weak-subjectivity-period).
```
Comment (raxhvl) [inline docs/wiki/CL/syncing.md:14]:
```suggestion
With the merge, execution blocks are wrapped inside beacon blocks (the beacon chain existed in parallel to the main chain, hence "The Merge") and clients are split into three: an execution layer client, a consensus layer client, and optionally a validator client. Due to this change, the CL node depends on the EL node to verify the execution payload within the beacon blocks. For syncing, however, the EL node depends on the CL node for a sync target. It then follows the same steps as an ETH1 client except. To provide this target, the CL node needs to sync the beacon chain first and it ***CANNOT*** use the same approach as the EL node, i.e., to simply sync from genesis. In fact, if you have ever run a CL node, it would complain that syncing from genesis is [unsafe](https://docs.teku.consensys.io/concepts/weak-subjectivity#sync-outside-the-weak-subjectivity-period).
```

**Review by raxhvl — COMMENTED** [review]
This is well written. I have left my review. Thanks!

File: docs/wiki/CL/syncing.md (hunk 1 above)
Comment (chirag-parmar) [reply docs/wiki/CL/syncing.md:14]: ah I think I left a word in there while editing

File: docs/wiki/CL/syncing.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+> [Subjective](https://dictionary.cambridge.org/dictionary/english/subjective): influenced by or based on personal beliefs or feelings, rather than based on facts  
+> [Objective](https://dictionary.cambridge.org/dictionary/english/objective): based on real facts and not influenced by personal beliefs or feelings (scroll down on the link)
+
+## Background
+
+Before the merge, a new node synced (configured for a "full sync") with the blockchain as described below:
```
Comment (taxmeifyoucan) [inline docs/wiki/CL/syncing.md:6]: First, you should explain there are different strategies for EL and CL sync. What you are describing here is a single strategy for EL sync which still works today, there is no before/after the Merge. There is snap sync, fast sync, pivot sync for EL. CL weak subjectivity worked even before the Merge. The only new post Merge sync strategy is optmistic sync

File: docs/wiki/CL/syncing.md (hunk 1 above)
Comment (taxmeifyoucan) [inline docs/wiki/CL/syncing.md:14]:
```suggestion
With the Merge, execution blocks are wrapped inside beacon blocks (the beacon chain existed in parallel to the main chain, hence "The Merge") and clients are split into three, an execution layer client, a consensus layer client, and optionally a validator client. Due to this change, the CL node depends on the EL node to verify the execution payload within the beacon blocks. For syncing, however, the EL node depends on the CL node for a sync target. It then follows the same steps as an ETH1 client except. To provide this target, the CL node needs to sync the beacon chain first and it ***CANNOT*** use the same approach as the EL node, i.e., to simply sync from genesis. In fact, if you have ever run a CL node, it would complain that syncing from genesis is [unsafe](https://docs.teku.consensys.io/concepts/weak-subjectivity#sync-outside-the-weak-subjectivity-period).
```

File: docs/wiki/CL/syncing.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+> ETH1 backfills headers but the verification of block bodies happens from genesis -> target. In ETH2, the verification itself happens from target -> genesis. The core of blockchain syncing is verification; without a verified syncing process, there is no point in syncing. Hence, the verification direction is usually considered the syncing direction.
+
+The second difference lies in the anchor of trust in information. Since the history of the chain can be changed under certain conditions the sync target cannot be objectively verified i.e. we can never concretely prove its existence in the canonical chain. Hence, there is a significant amount of trust placed in the sync target and it remains subjective. Because of the trust involved, sync targets (named weak subjectivity checkpoints) are shared out-of-band since random peer connections in the p2p layer of Ethereum cannot be trusted. It is important to note that, only the sync target remains subjective, all other information obtained is objectively verified i.e. the chain is *weakly* subjective.
```
Comment (taxmeifyoucan) [inline docs/wiki/CL/syncing.md:24]: An article on week subjectivity should dive into finality and how it keeps the chain secure. 

Btw there is a decent article on ethorg docs but I suppose this page should focus more on the node experience and client behavior under the hood https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/weak-subjectivity/

File: docs/wiki/CL/syncing.md (hunk 2 above)
Comment (chirag-parmar) [reply docs/wiki/CL/syncing.md:6]: This seems to be the main point of contention for everyone who has proof read the document. Everything mentioned in the comment above is true, but snap sync, fast sync and pivot sync are just optimizations over the definitive method of syncing. Moreover, every client team in the past and present has had varying definitions of the mentioned optimizations. Mentioning them will serve no (as far as my knowledge goes) extra relevant information when considering weak subjectivity and other changes in the syncing process seen between a PoW Ethereum and a PoS Ethereum.  For this, I think I should remove the sidenote "configured for a full sync" to avoid any confusion.

Regarding the chronology, whether CL weak subjectivity worked before the merge or not is irrelevant in the context of the canonical chain, IMHO. The only reason for explicitly mentioning "Before the Merge" and "After the Merge" is for the Merge to serve as a vantage point that brought the change in consensus mechanisms and also its effect on the syncing process. I think I should rephrase it as "Before the canonical chain switched to POS".

Do you think this would resolve the contention?

File: docs/wiki/CL/syncing.md (hunk 3 above)
Comment (chirag-parmar) [reply docs/wiki/CL/syncing.md:24]: I think you nailed it on this one. That article indeed touched everything related to weak subjectivity, therefore, I also linked it in the footer. It is a little opaque and assumes context of subjectivity and objectivity in blockchain technology which is the gap I wanted to fill. Vitalik's blog on the matter is too abstract for EPS fellows to consume, I think. 

I think I touched on how weak subjectivity keeps the chain secure(maybe I did not fully?) but I'm not sure what role it has in finality. Maybe I am missing information. Can you point me to the specific stuff, I think my brain is fogged out XD?

File: docs/wiki/CL/syncing.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+4. *Optimistically follow the head of the chain and continuously update the target header for the execution chain*
+5. *Once the EL is synced, then mark the CL slots as verified post-verification. The node may now be considered fully synced*
+
+### Summary
+
+1. Socially sourcing a piece of information doesn't make the chain subjective. Ironically, because the checkpoints remain subjective they are shared socially. It is the trust put it in the information that makes the chain subjective. 
+2. Optimistic Sync and Checkpoint Sync cannot be considered different modes of syncing. They solve the same problem but at different stages of the syncing process. Moreover, you need both the modes of syncing to reliably sync the chain. 
+3. The paradigm shift in the syncing process eludes most: beacon goes future -> past(backfilling) and execution goes past -> future(genesis syncing). The direction is determined by the direction of verification.
+
+## Useful Links
```
Comment (taxmeifyoucan) [inline docs/wiki/CL/syncing.md:41]: some more resources, mostly historical reference 
https://blog.ethereum.org/2014/11/25/proof-stake-learned-love-weak-subjectivity
https://notes.ethereum.org/@adiasg/weak-subjectvity-eth2
https://blog.ethereum.org/2016/05/09/on-settlement-finality

File: docs/wiki/CL/syncing.md (hunk 2 above)
Comment (taxmeifyoucan) [reply docs/wiki/CL/syncing.md:6]: Yes I agree, no need to dive into specific strategies for EL but you should add context that there are variations of EL sync strategy, make a clear distinction between EL and CL here.

> remove 'configured for a full sync'

What you describe below is the naive full sync strategy so it shouldn't be removed. 

> "Before the canonical chain switched to POS"

That is literally the same as saying 'before the Merge.' But it the same as today, EL syncs very same way you are describing. I would rephrase this to make it clear that this is how EL handles finality with PoW or PoS

File: docs/wiki/CL/syncing.md (hunk 3 above)
Comment (taxmeifyoucan) [reply docs/wiki/CL/syncing.md:24]: A recent finalized block becomes a weak subjectivity checkpoint. It should be clear that by finalizing the chain Casper is basically providing weak subjectivity to sync

File: docs/wiki/CL/syncing.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+> [Subjective](https://dictionary.cambridge.org/dictionary/english/subjective): influenced by or based on personal beliefs or feelings, rather than based on facts  
+> [Objective](https://dictionary.cambridge.org/dictionary/english/objective): based on real facts and not influenced by personal beliefs or feelings (scroll down on the link)
+
+## Background
+
+Before the merge, a new node synced (configured for a "full sync") with the blockchain as described below:
+1. *Query for the latest "final" block from other nodes in the network.* In ETH1, there was no finality as such, but reversing the "work" (Proof-of-work) of 6 blocks was considered difficult enough to call the 7th block from the head "final."
```
Comment (taxmeifyoucan) [inline docs/wiki/CL/syncing.md:7]: One more thing which repeats in the whole doc - you keep using 'ETH1' here which is wrong nomenclature. Please keep it as EL/CL client to be consistent with rest of the content

File: docs/wiki/CL/syncing.md (hunk 2 above)
Comment (chirag-parmar) [reply docs/wiki/CL/syncing.md:6]: Ah I see what you mean now. I'll change this. Seems like I just needed a fresh perspective

File: docs/wiki/CL/syncing.md (hunk 5 above)
Comment (chirag-parmar) [reply docs/wiki/CL/syncing.md:7]: Ah, really gotta change that habit. Thanks

File: docs/wiki/CL/syncing.md (hunk 3 above)
Comment (chirag-parmar) [reply docs/wiki/CL/syncing.md:24]: Thanks for pointing this out, usually obvious information is missed out making it difficult for someone just starting with the tech to understand.

**Comment by chirag-parmar** [comment]
@taxmeifyoucan yes, apologies it got super busy the past few months. I'll wrap this up by the end of the week

**Comment by taxmeifyoucan** [comment]
Thank you Chirag, this looks great!
